plugins/modules/device_credential_intent.py

```diff
@@ -1987,173 +1966,185 @@ def get_want_assign_credentials(self, AssignCredentials):
-
-            # All CLI details from the Cisco DNA Center
-            cli_details = global_credentials.get("cliCredential")
-            if not cli_details:
-                self.msg = "No Global CLI Credential is available"
-                self.status = "failed"
-                return self
-            cliDetail = None
-            if cliId:
-                cliDetail = get_dict_result(cli_details, "id", cliId)
-                if not cliDetail:
-                    self.msg = "CLI credential ID is invalid"
-                    self.status = "failed"
-                    return self
-            elif cliDescription and cliUsername:
-                for item in cli_details:
-                    if item.get("description") == cliDescription and \
-                            item.get("username") == cliUsername:
-                        cliDetail = item
-                if not cliDetail:
-                    self.msg = "CLI credential username and description is invalid"
-                    self.status = "failed"
-                    return self
-            want.get("assign_credentials").update({"cliId": cliDetail.get("id")})
-
-        snmpV2cReadId = AssignCredentials.get("snmp_v2c_read").get("id")
-        snmpV2cReadDescription = AssignCredentials.get("snmp_v2c_read").get("description")
-        if snmpV2cReadId or snmpV2cReadDescription:
-
-            # All snmpV2cRead details from the Cisco DNA Center
-            snmpV2cRead_details = global_credentials.get("snmpV2cRead")
-            if not snmpV2cRead_details:
-                self.msg = "No Global snmpV2cRead Credential is available"
-                self.status = "failed"
-                return self
-            snmpV2cReadDetail = None
-            if snmpV2cReadId:
-                snmpV2cReadDetail = get_dict_result(snmpV2cRead_details, "id", snmpV2cReadId)
-                if not snmpV2cReadDetail:
-                    self.msg = "snmpV2cRead credential ID is invalid"
-                    self.status = "failed"
-                    return self
-            elif snmpV2cReadDescription:
-                for item in snmpV2cRead_details:
-                    if item.get("description") == snmpV2cReadDescription:
-                        snmpV2cReadDetail = item
-                if not snmpV2cReadDetail:
-                    self.msg = "snmpV2cRead credential username and description is invalid"
+        cli_credential = AssignCredentials.get("cli_credential")
+        if cli_credential:
+            cliId = cli_credential.get("id")
+            cliDescription = cli_credential.get("description")
+            cliUsername = cli_credential.get("username")
+            if cliId or cliDescription and cliUsername:
+
+                # All CLI details from the Cisco DNA Center
+                cli_details = global_credentials.get("cliCredential")
+                if not cli_details:
+                    self.msg = "No Global CLI Credential is available"
```

Comment:
self.msg = "Global CLI credential is not available"
or
    self.msg = "Global CLI credential is not found"